src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java

```diff
@@ -109,16 +110,41 @@ public RefundListDtoResponse getRefundList(String status, MultiValueMap<String,
             RefundStatus refundStatus = RefundStatus.getRefundStatus(status.toLowerCase());
             //get the refund list except the self uid
             refundList = SENTFORAPPROVAL.getName().equalsIgnoreCase(status) && "true".equalsIgnoreCase(
-                excludeCurrentUser) ? refundsRepository.findByRefundStatusAndCreatedByIsNot(
-                refundStatus,
-                uid
+                    excludeCurrentUser) ? refundsRepository.findByRefundStatusAndCreatedByIsNot(
+                    refundStatus,
+                    uid
             ) : refundsRepository.findByRefundStatus(refundStatus);
         }
 
+        // Filter Refunds List based on Service Type
+        if (null != roles && !roles.isEmpty()) {
+            refundList = filterRefundList(refundList, headers, roles);
+        }
+
         return getRefundListDto(headers, refundList);
     }
 
+    private Optional<List<Refund>> filterRefundList(Optional<List<Refund>> refundList, MultiValueMap<String, String> headers,
+                                                    List<String> roles) {
+        Set<String> distintUserIDSet = idamService.getUserIdSetForService(headers, roles);
+
+        /*refundList.ifPresent(list -> list.stream()
+                .filter(refunds -> distintUserIDSet.contains(refunds.getCreatedBy()))
+                .collect(Collectors.toList()));*/
+
+        List<Refund> finalList = new ArrayList<>();
```

Comment:
Suggestion: Change this to stream if possible

Reply:
Done